splunk-otel-android/src/main/java/io/opentelemetry/rum/internal/instrumentation/activity/ActivityTracer.java

```diff
@@ -115,38 +119,96 @@ private Span createSpanWithParent(String spanName, @Nullable Span parentSpan) {
         final SpanBuilder spanBuilder =
                 tracer.spanBuilder(spanName)
                         .setAttribute(ACTIVITY_NAME_KEY, activityName)
-                        .setAttribute(SplunkRum.COMPONENT_KEY, SplunkRum.COMPONENT_UI);
+                        .setAttribute(COMPONENT_KEY, COMPONENT_UI);
         if (parentSpan != null) {
             spanBuilder.setParent(parentSpan.storeInContext(Context.current()));
         }
         Span span = spanBuilder.startSpan();
         // do this after the span is started, so we can override the default screen.name set by the
         // RumAttributeAppender.
-        span.setAttribute(SplunkRum.SCREEN_NAME_KEY, screenName);
+        span.setAttribute(SCREEN_NAME_KEY, screenName);
         return span;
     }
 
-    void endSpanForActivityResumed() {
+    public void endSpanForActivityResumed() {
         if (initialAppActivity.get() == null) {
             initialAppActivity.set(activityName);
         }
         endActiveSpan();
     }
 
-    void endActiveSpan() {
+    public void endActiveSpan() {
         // If we happen to be in app startup, make sure this ends it. It's harmless if we're already
         // out of the startup phase.
         appStartupTimer.end();
         activeSpan.endActiveSpan();
     }
 
-    ActivityTracer addPreviousScreenAttribute() {
+    public ActivityTracer addPreviousScreenAttribute() {
         activeSpan.addPreviousScreenAttribute(activityName);
         return this;
     }
 
-    ActivityTracer addEvent(String eventName) {
+    public ActivityTracer addEvent(String eventName) {
         activeSpan.addEvent(eventName);
         return this;
     }
+
+    public static Builder builder(Activity activity) {
+        return new Builder(activity);
+    }
+
+    static class Builder {
+        private final Activity activity;
+        private AtomicReference<String> initialAppActivity = new AtomicReference<>();
+        private Tracer tracer;
+        private AppStartupTimer appStartupTimer;
+        private ActiveSpan activeSpan;
+
+        public Builder(Activity activity) {
+            this.activity = activity;
+        }
+
+        public Builder visibleScreenTracker(VisibleScreenTracker visibleScreenTracker) {
```

Comment:
We usually use the `set...` naming scheme in builders

Reply:
Eww. Ok, I'll use setters.